crochetcraft/src/routes/tools/curveBuilder/+page.svelte

```diff
@@ -0,0 +1,198 @@
+<script lang="ts">
+    import { page } from '$app/stores';
+    import ThreeCanvas from '$lib/ThreeCanvas.svelte';
+    import PointsTable from '$components/PointsTable.svelte';
+    import { onMount } from 'svelte';
+    import * as THREE from 'three';
+    import { Vector3 } from 'three';
+    import { CubicSpline3D } from '$lib/geometry/cubicSpline';
+    import { placeDebugSpheres } from '$lib/render/debug';
+    import { vectorListToCode } from '$lib/builder/codegen';
+
+    function enumToKeys(type: object) {
+        return Object.values(type).filter((x) => !isNaN(x as number));
+    }
+
+    enum DrawType {
+        Point,
+        Line,
+        Cylinder,
+    }
+    const drawTypeOptions = enumToKeys(DrawType);
```

Comment:
Hmm, this `enumToKeys` is a little ugly IMO. Personally, I prefer to do this:

```ts
enum DrawType {
    Point = 'Point',
    Line = 'Line',
    Cylinder = 'Cylinder',
}
Object.keys(DrawType) // => ['Point', 'Line', 'Cylinder']
```

I also like this style because it's easier to debug (e.g., if you `console.log` an enum value, you'll get `'Point'` instead of `1`). Of course, the downside is that this is repetitive, since each entry in the enum needs to be repeated.

@simcard0000 Thoughts?

Reply:
I think I did this to get around repetition jank (makes the input/settings logic slightly nicer, since it doesn't cause settings to go from integer to string type)

Reply:
What do you mean by that?

Reply:
cc @j-tai I agree with Jasmine that string enums make things easier to debug, but what would be a situation where it would be much more advantageous to have a numeric enum (i.e. what Allen originally has)

Reply:
Object.keys produces `['Point', 'Line', 'Cylinder', 0, 1, 2]`
The `enumKeys` helper filters out the duplicates. I choose to filter out the strings since Svelte bindings default to using the numbers.